packages/sdk/src/energyCost.ts

```diff
@@ -60,6 +68,42 @@ export function getEnergyCost(
     );
 }
 
+/**
+ * Get contract update energy cost
+ * Estimated by calculateEnergyCost, where transactionSpecificCost received from invokeContract used energy
+ * @param {ConcordiumGRPCClient} grpcClient - The client to be used for the query
+ * @param {ContractAddress.Type} contractAddress - The address of the contract to query
+ * @param {AccountAddress.Type} invoker - Representation of an account address
+ * @param {Parameter.Type} parameter - Input for contract function
+ * @param {ReceiveName.Type} method - Represents a receive-function in a smart contract module
+ * @param {bigint} signatureCount - Number of expected signatures
+ */
```

Comment:
**Suggestion:** Please document when this function throws and also that the estimate is given for the last finalized block according to the node, which means this the actual energy cost might be different depending on the implementation of the smart contract and the interaction with the instance, since this was estimated.

Reply:
Would it not make sense to add an optional `blockHash` argument which can be passed to the grpc client?